src/main/java/io/pravega/perf/WriterWorker.java

```diff
@@ -10,39 +10,57 @@
 
 package io.pravega.perf;
 
+import javax.annotation.concurrent.NotThreadSafe;
 import java.io.IOException;
-import java.time.Duration;
-import java.time.Instant;
 import java.util.Random;
 import java.util.concurrent.Callable;
-import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ExecutionException;
 
 /**
  * abstract class for Writers.
  */
 public abstract class WriterWorker extends Worker implements Callable<Void> {
+    final private static int MS_PER_SEC = 1000;
+
     final private Performance perf;
-    final private ThroughputController tput;
+    final private EventsController eCnt;
+    final private String payload;
 
     WriterWorker(int sensorId, int events, int secondsToRun,
-                 boolean isRandomKey, int messageSize, Instant start,
-                 PerfStats stats, String streamName, ThroughputController tput) {
+                 boolean isRandomKey, int messageSize, long start,
+                 PerfStats stats, String streamName, int eventsPerSec, boolean wNr) {
 
         super(sensorId, events, secondsToRun,
-                isRandomKey, messageSize, start,
-                stats, streamName, 0);
-        this.tput = tput;
-        perf = secondsToRun > 0 ? new ThroughputWriter() : new EventsWriter();
+                messageSize, start, stats,
+                streamName, 0);
+        this.eCnt = new EventsController(start, eventsPerSec);
+        perf = secondsToRun > 0 ? (wNr ? new EventsWriterTimeRW() : new EventsWriterTime()) :
+                (wNr ? new EventsWriterRW() : new EventsWriter());
+
+        Random random = new Random(0);
+        byte[] bytes = new byte[messageSize];
+        for (int i = 0; i < messageSize; ++i) {
+            bytes[i] = (byte) (random.nextInt(26) + 65);
+        }
+        payload = new String(bytes);
```

Comment:
Specify the character encoding so this does not generate a warning.

Reply:
its changed to US_ASCII.